### PR TITLE
The Ballistics Expansion

### PR DESCRIPTION
**Pull request by Skrem7**
# Document the changes in your pull request

For the sake of conciseness, and because it would take a literal essay to explain these changes, I'm just going to list what's done. This also actually utilizes all the shit done in #14486

- Adds LWT-650 DMR to the NT uplink for 10 WC. It fires .308.
- .308 magazines standardize to 15 bullets and do 42 damage with -40 wound bonus. They have no wound_drop_tile value. An additional magazine costs 2 WC.
- .308 Penetrator acts as DU slugs, penetrating through all mobs. It does 35 damage with 35 AP. Costs 4 WC.
- .308 Heavy Laser fires a heavy laser, of the same projectile as the Solaris, I believe (40 damage, ignite, good wound chance). Can also recharge like a laser rifle magazine. Costs 7 WC.
- Sprites and sounds for the LWT-650 DMR and all of its ammunition have been added
- .308 and 7.76 now have a speed of 0.3, just like sniper rounds
- Adds 5.56 AP, which does 31 damage with 40 AP. It costs 6 TC/WC.
- Adds 5.56 Incendiary, which 23 damage with 2 fire stacks. It costs the standard 4 TC/WC.
- Adds 5.56 Rubber, which does 10/45 damage/stamina. It costs the standard 4 WC.
- 5.56 toploader magazines have progressive sprites
- Makes it so M-90gl and NT ARG use different magazine types and adds respective sprites for the differing magazines and their alternative ammunitions
- As a statement, the main difference between .45 and 10mm is that .45 has a -10 wound bonus. 10mm has none.
- Adds .45 AP, which does 27 damage with 40 AP. Cost is 4 TC for nukies.
- Adds .45 HP, which does 45 damage with -45 AP. It is also SHARP_EDGED, possessing a -15 wound bonus with a +5 bare_wound_bonus. 10mm HP has also received the SHARP_EDGED, though its wound/bare bonuses are -25/+5. Cost for the .45 variant is the standard 3 TC for nukies.
- Adds .45 Venom, which does 20 damage while injecting 4 units of venom to carbon targets hit. Cost is 4 TC for nukies.
- Adds new sprites for all these .45 variants
- Adds 10mm EMP which does 20 damage and a one-tile **weak** EMP. A single mag costs 1 TC. Adds its sprites too.
- Updates all existing sprites for 10mm and 9mm alternative ammos.
- Re-introduces Frag-12 rounds for the Bulldog. They do 25 damage on hit, no longer paralyze, but emit a light explosion at a radius of 2. They cost 3 TC for nukies.
- HP Buckshot has been removed. In its place is Syndicate buckshot, which deals 18 damage per pellet with 2/2 wound/bare bonuses, as opposed to the new 12 damage per pellet with 5/5 of new standard buckshot, up from 11 damage per pellet. Any 12g magazine unless otherwise stated costs 2 TC, as before.
- Rubber shot now deals 14.5 stamina damage per pellet instead of 13.
- Flechette has had its wound chance brought up overall, it also possesses less value falloff from range. It costs 3 TC.
- DU slugs now deal 26 damage from 30, but wound bonus has been increased slightly
- Cryo shot temperature adjustment now ignores armor
- Improv slugs, assuming they hit at range, will suck less
- All shotgun slugs have been put under the same parent /slug/, which now has a projectile speed of 0.5. All pellets also have a speed of 0.5 instead of the default 0.4 of bullets.
- .32 TRAC implant only lasts 5 minutes now, down from 15
- EMPs now removes .32 TRAC implants
- Caldwell Revolver and its ammo can no longer be researched and printed
- Removes TRAC speedloaders from Tracking Implant crate
- Adds new Tracking Revolver crate for 4000 which contains 1 Caldwell and 2 speedloaders for it
- Magazines that go into rechargers such as the laser rifle's now charge faster if the recharger is upgraded
- Occupying Force WT no longer has a contradicting description nor a unique name, just spawns with rubber rounds. This means the item could be used as a roundstart armory WT, if so desired in a future PR.
- Fixes an oversight on 4.6x30mm incendiary inherit which meant it did not retain the 20 AP of the standard bullet
- Surplus carbine (aka 10mm surplus rifle) has had its fire delay decreased to 12 from 30.
- .357 and .50ae fire delays have been increased to 6 and 8 respectively. Special .357 revolvers remain unaffected. Nagant revolver now has 5 fire delay. Caldwell now has a fire delay of 10. All others have a fire_delay of 0.
- .38 reinforcement now make the fire_delay 12 (as opposed to 6 of standard .357) and spread of 25, but no longer has a chance to blow up in your hand
- .357 Metalshock now properly does a tesla arc, instead of doing an internal shock and confusion
- .357 Wallstake now costs 3 TC
- Adjusts code to be shorter and include inherits all around regarding ammunition, including spriting code which means several redundant sprites have been removed
- Bulldog has had its gun sprite reworked, now gives a visual indicator of ammo left rather than a snowflake color if you load an alternative mag
- Pre-existing bulldog alternative magazines that lacked sprites now have them
- Adds new load, unload, rack, and fire sounds to accompany the LWT-650 DMR. All but the fire sound could potentially be utilized for other rifles.
- Every single ballistic magazine name and description has received a passover to make sure a specific examine actually existed. Several names and descriptions overall relating to guns have been adjusted, including WT-550 Rifle -> WT-550 Carbine, M-90gl Carbine -> M-90gl Rifle, and Surplus Rifle -> Surplus Carbine

# Spriting
New LWT-650 with mag:
image

New bulldog progressive sprites:
Capacity = 100% -> 61%
image
Capacity = 60% -> 21%
image
Capacity = 20% -> 1%
image
Capacity = 0% (The light is blinking)
image

10mm AP: 
image

10mm HP: 
image

10mm Soporific: 
image

10mm Incendiary: 
image

9mm AP: 
image

9mm Incendiary: 
image

10mm EMP:
image

.45 AP:
image

.45 HP:
image

.45 Venom:
image

5.56 Casing: 
image

5.56 (Toploader) AP:
image

5.56 AP Casing: 
image

5.56 (Toploader) Incend:
image

5.56 Incend Casing: 
image

5.56 (ARG) Standard:
image

5.56 (ARG) AP: 
image

5.56 (ARG) Incend:
image

5.56 (ARG) Rubber:
image

5.56 Rubber Casing: 
image

.308 Standard: 
image

.308 Penetrator: 
image

.308 Penetrator Casing: 
image

.308 Heavy Laser (Red lines flash when empty): 
image

.308 Heavy Laser Casing: 
image

Syndicate Buckshot: 
image

Syndicate Slug:
image

12-gauge slug:
image

12-gauge frag:
image

12-gauge flechette:
image

# Wiki Documentation

Sprites and values of LWT-650 DMR and 5.56/.45/shotgun shell alternative ammos/10mm/9mm to be uploaded to Guide to Combat, Syndicate Items, Nanotrasen Items, plus new TC/WC costs
Shotgun shells to receive new values
Metalshock to receive an updated description
Names of every instance of WT-550, Surplus Rifle, and M-90gl to be changed to their proper form
TRAC revolver crate to be added to list of crates, tracking implant crate adjusted
.38 reinforcement mechanics to be addendum'd on the Detective page and Guide to Combat
.45 and 12 gauge ammo bag for nukies to receive new items
TRAC revolver and its ammo to be removed from R&D and clarified for obtaining in Security items. TRAC timer needs to be adjusted if it is stated anywhere (Guide to Combat, Security Items?)
HP buckshot is gone and so is its drum mag; new standard buckshot for the Bulldog is filled with Syndicate buckshot

# Changelog

:cl: Skrem_7, @KillerOrcaCora for LWT sprites, @MajManatee for the initial rifle concept and work, @Mqiib for collaboration of some balance pre-draft, a lot of coffee and spite
experimental: Begin log of Ballistic Expansion.
rscadd: Adds the LWT-650 DMR, a marksman rifle available to ERTs that fires a new .308 round, most like a weaker version of 7.76, strictly damage-speaking.
rscadd: Adds .308 Penetrator and Heavy Laser for the ERT rifle.
rscadd: Adds 5.56 AP, Incendiary, and Rubber. AP and Incendiary are purchasable by nukies, while ERT can buy them all. 
rscadd: The M-90gl and the NT ARG 'Boarder' Rifle now have different magazine variants.
rscadd: Adds .45 AP, HP, and Venom for the C-20r which nukies can buy.
rscadd: Adds 10mm EMP which does a one-tile EMP on the target hit.
rscadd: Frag rounds exist for nukies again, this time they blow up 2 tiles in a weak explosion but they do NOT paralyze.
rscadd: Standard ammo of the Bulldog and the magazine bought is now Syndicate buckshot, which does a lot more damage
rscadd: You can now order the Caldwell and two speedloaders for it from armory
rscdel: Removes HP Buckshot
rscdel: Caldwell and its ammo can no longer be researched and printed
rscdel: Tracking implant crate only contains a crate of tracking implants again
tweak: 7.76 and the new .308 have 0.3 speed like the sniper
tweak: .32 TRAC now only lasts 5 minutes
tweak: .32 TRAC implants are now destroyed by EMPs
tweak: Magazines such as the laser rifle's and the new .308 Heavy Laser now charge faster if the recharger is upgraded
tweak: .45 ammo duffle bag now has 5 standard mags, 2 ap mags, and 2 hp mags
tweak: 12g ammo duffle bag now has 3 syndie buckshot mags, 3 syndie slug mags, and 2 flechette mags
tweak: Occupying Force special WT has been re-branded as a standard WT, it just starts with rubber rounds. It's now a preferred item to be utilized if WT's were to ever spawn in armory roundstart.
tweak: Surplus carbine (or 10mm surplus rifle) has a fire delay of 12 instead of 30 now.
tweak: Syndicate revolver fire_delay has been increased to 6 (includes Russian Revolver), modified .38 fire_delay is 12, Deagle fire_delay is 8. Nagant revolver fire_delay is 5. The .38 and special .357 revolvers have a fire_delay of 0 again. TRAC revolver fire_delay adjusted to 10.
tweak: .38 modified no longer blows up in your hands, but has inaccuracy
tweak: .357 Metalshock now properly causes a tesla arc that can chain that will HURT
tweak: .357 Wallstake now costs 3 TC.
tweak: 10mm HP adjusted to be sharp-edged. Its wound bonuses have been decreased to compensate for this.
tweak Some shotgun shells, mostly ones that fire pellets, have received a passover in regards to their values and effects. All ballistic slugs and pellets now have a speed of 0.5, instead of 0.4
tweak: Almost all bullets have had their inherit and sprite code passed over to shorten their length and be more robust in quality
tweak: 4.6x30mm incend has AP of the normal now because rubber inherits it but incend hasn't because of its pathing
soundadd: Adds rifle loading, rifle unloading, and rifle racking sounds taken from TGMC. DMR sound which was ripped from TGMC which was ripped from Halo: Reach has also been added
imageadd: Adds new sprites for 5.56 alternative ammos, NT 'ARG' Boarder magazine and alternative magazine sprites, alternative .45 C-20r magazines, Syndicate Buckshot and Syndicate slug sprites, 12g drum mag sprites for each ammo type that did not have it, new 5.56 casing types including types of alternative ammunition, .308 magazines and their alternatives, and the 10mm EMP pistol magazine as well as updating all existing 10mm and 9mm alternative ammo sprites
imageadd: 5.56 Toploader magazines now have dynamic sprite_states based on the level of ammunition
imageadd: The gun sprite for bulldog now properly reflects the level of ammunition you have left in the magazine as well as being standardized across ammo types
imagedel: Deletes several redundant in-hand ammo images due to how magazines initialize with update_icon()
imagedel: HP Buckshot is gone
imagedel: Bulldog shotgun no longer has snowflake sprites for varying magazines being loaded
spellcheck: Adds or adjusts a metric LOAD of ballistic-related descriptions
experimental: End log of Ballistic Expansion.
/:cl:

### CODE REVIEW

**Review by KillerOrcaCora — DISMISSED** [review]
Remove sprites from this PR for a review into smaller digestible sizes based on gun+ammo.

**Comment by cuackles** [comment]
> Remove sprites from this PR for a review into smaller digestible sizes based on gun+ammo.

is is illogical, the sprites are needed to add the new items, removing them would just delay everything, the sprites are displayed correctly and by my review they are all acceptable, ammo sprites are correctly designated and each one of them is recognizable, gun sprites are acceptable though i find a bit odd that the LWT which is a CC gun is red, but if i consider it a big issue i will fix it in the future, but apart from that, they are good, same for inhand sprites, adding to that your sprites are in this pr so you shouldn't  be approving or denying it in any case, anyway sprites good so it gets art approval and I'm dismissing orcas change request cus it aint making much sense

**Review by cuackles — APPROVED** [review]

**Comment by Marmio64** [comment]
way too much stuff for a single PR, mostly i dont like EMPs being even more stupidly easy for traitors to use by giving them ammo to EMP people in one shot without need for an EMP nade that takes time to pin and can also disable your own equipment if you brought any.

**Comment by Skrem7** [comment]
> way too much stuff for a single PR, mostly i dont like EMPs being even more stupidly easy for traitors to use by giving them ammo to EMP people in one shot without need for an EMP nade that takes time to pin and can also disable your own equipment if you brought any.

image
Yes, this is melee but this is re-usable, metashielded, camouflaged, instant, and silent.

**Comment by Mqiib** [comment]
> way too much stuff for a single PR, mostly i dont like EMPs being even more stupidly easy for traitors to use by giving them ammo to EMP people in one shot without need for an EMP nade that takes time to pin and can also disable your own equipment if you brought any.

There is literally no way to give them "easier" EMPs, they get an implant that can be activated while stunned with a massive AOE for 2 tc PLUS a box of EMP grenades. It's not a big deal.

**Comment by JamieD1** [comment]
Jesus christ.

**Comment by Skrem7** [comment]
> Jesus christ.

If you desperately want me to I could split it into the following PRs, but merging one after the other is going to cause a shitton of conflicts and I wouldn't be happy doing it, but I also understand trying to maintain and review a PR of this size is absolute misery. If it's of any gracious aid, most of these changes I was actively in a voice call with Mqiib going over and testing.

1) Add the LWT-650 and its respective ammo/sounds/images, also make 7.76 slightly faster
2) Alternative 5.56 and its new sprites for the Boarder, too
2.1 or 3) Alternative .45 and its new sprites
2.2 or 3.1 or 4) 10mm EMP
4) Revolver passovers (basically firerate, .357 ammos, and .38 modified)
5) Shotgun ammo and sprite passovers, including its speed
6) TRAC nerf/adjustment
7) All-around description, name, or sprite additions/tweaks
8) WT incend AP
9) Surplus rifle fire rate up

**Comment by Mqiib** [comment]
Can confirm, most of these changes were run by me, they SHOULD be okay unless anyone sees anything particularly egregious. only thing I'm a little concerned about is the balance of the 650 since it has really high damage but it shouldn't be too much of a problem, probably.

File: code/modules/projectiles/boxes_magazines/external/smg.dm [hunk 1]
```diff
@@ -95,48 +63,79 @@
...
+	name = "SMG magazine (Armor-Piercing .45)"
+	desc = "A 24-round magazine for the C-20r SMG that contains AP .45 rounds. \
+			These rounds do less damage but bypass most standard body armor."
+	icon_state = "c20r45A-24"
+	ammo_type = /obj/item/ammo_casing/c45/ap
+	sprite_designation = "A"
+
+/obj/item/ammo_box/magazine/smgm45/hp
+	name = "SMG magazine (Hollow-Point .45)"
+	desc = "A 24-round magazine for the C-20r SMG that contains AP .45 rounds. \
```
Comment (Skrem7) [inline code/modules/projectiles/boxes_magazines/external/smg.dm]:
```suggestion
	desc = "A 24-round magazine for the C-20r SMG that contains HP .45 rounds. \
```

File: code/modules/projectiles/boxes_magazines/external/smg.dm [hunk 2]
```diff
@@ -95,48 +63,79 @@
...
+			These rounds suffer against those wearing body armor but devastate those who do not."
+	icon_state = "c20r45H-24"
+	ammo_type = /obj/item/ammo_casing/c45/hp
+	sprite_designation = "H"
+
+/obj/item/ammo_box/magazine/smgm45/venom
+	name = "SMG magazine (Venom .45)"
+	desc = "A 24-round magazine for the C-20r SMG that contains AP .45 rounds. \
```
Comment (Skrem7) [inline code/modules/projectiles/boxes_magazines/external/smg.dm]:
```suggestion
	desc = "A 24-round magazine for the C-20r SMG that contains Venom .45 rounds. \
```

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**